freeipa/src/main/java/com/sequenceiq/freeipa/service/CredentialService.java

```diff
@@ -44,4 +53,16 @@ public Credential getCredentialByEnvCrn(String envCrn) {
         String attributes = secretService.getByResponse(secretResponse);
         return new Credential(credentialResponse.getCloudPlatform(), credentialResponse.getName(), attributes, credentialResponse.getCrn());
     }
+
+    @Override
+    public String getCloudPlatformByCredential(String credentialName, String accountId, CredentialType type) {
```

Comment:
this method and the implementation seems to be duplicated several times. I think it should be refactored, maybe to a default implementation in the interface or similar

Reply:
I cannot create a common implementation, because we need implementation in every microservice. If I create a default implementation in the env-api, the env service won't start (and every other microservice where a custom impl will be created).

But I think I will do it another way. The annotation is nicer but more complicated because the env network validator should retrieve the credential because of the cloud platform. And If I looking for a common way of implementation, it will increase the complexity of the whole task exponentially. `@NotNull` annotation could not be used, because the yarn provider is accepted without a network and the qaas test will fail.